src/bin/hexyl.rs

```diff
@@ -161,10 +178,124 @@ fn main() {
     }
 }
 
-fn parse_hex_or_int(n: &str) -> Option<u64> {
-    if n.starts_with("0x") {
-        u64::from_str_radix(n.trim_start_matches("0x"), 16).ok()
-    } else {
-        n.parse::<u64>().ok()
+fn parse_byte_count(n: &str, block_size: u64) -> Option<u64> {
+    const HEX_PREFIX: &'static str = "0x";
+
+    let n = {
+        let mut chars = n.chars();
+        match chars.next()? {
+            '+' => chars.as_str(),
+            _ => n,
+        }
+    };
+
+    if n.starts_with(HEX_PREFIX) {
+        let n = &n[HEX_PREFIX.len()..];
+        if n.chars().next() == Some('+') {
+            return None;
+        }
+        return u64::from_str_radix(n, 16).ok();
+    }
+
+    let (n, unit_multiplier) = match n.chars().position(|c| !c.is_ascii_digit()) {
+        Some(unit_begin_idx) => {
+            let (n, raw_unit) = n.split_at(unit_begin_idx);
+            let raw_unit = Ascii::new(raw_unit);
```

Comment:
do we need a new dependency for this? Couldn't we just use `unit == raw_unit.to_lower()`?

Reply:
I'm used to approaching performance first, which was why I reached for `unicase` out of habit, but the fact is nobody will likely notice an allocation to get a lowercase string here. If you'd prefer that over another dep, sounds valid to me!

Reply:
I would prefer the easier solution without the extra crate. This will also make it easier for package maintainers (for distributions that package every dependency, like Fedora or Debian).

Reply:
I've removed `unicase` and now use `str::to_lowercase` instead.